goebpf_mock/mock_map.go

```diff
@@ -97,6 +97,22 @@ static void *bpf_find_item_by_key(struct __create_map_def *map, const void *key)
 	return item;
 }
 
+static void *bpf_find_next_item_by_key(struct __create_map_def *map, const void *key)
```

Comment:
i tried to implement proper testing for this, but eventually gave up. problem i had is that this linked list seems to be circular, so not sure how to know when we get the last element of the list - to return the proper value to the caller code.